cpmpy/solvers/gurobi.py

```diff
@@ -266,7 +266,7 @@ def transform(self, cpm_expr):
         cpm_cons = flatten_constraint(cpm_expr)  # flat normal form
         cpm_cons = reify_rewrite(cpm_cons, supported=frozenset(['sum', 'wsum']))  # constraints that support reification
         cpm_cons = only_bv_implies(cpm_cons)  # anything that can create full reif should go above...
-        cpm_cons = linearize_constraint(cpm_cons)  # the core of the MIP-linearization
+        cpm_cons = linearize_constraint(cpm_cons, supported=frozenset({"sum", "wsum", "sub", "min", "max","mul", "abs","pow","div"}))  # the core of the MIP-linearization
```

Comment:
- I cannot see a situation where keeping "sub" is better than just translating it to a "wsum". I'd drop it.
- We will also need a separate transformation for "min", "max", "abs".
- Could "mod" be missing?

Reply:
Gurobi supports `sub` in its API (see `_make_numexpr`) above so I thought it would be best to let Gurobi handle it themselves :)

Reply:
This is  gurobi-specific. If `sub` is not passed in `supported`, it will be transformed anyway.

Reply:
So the rationale behind `supported` is that anything that is natively supported by the solver. `mod` is not, so linearize will complain, which is ok.